### PR TITLE
Fix distutils deprecation

### PR DESCRIPTION
**Pull request by ionelmc**
Ref #97

### CODE REVIEW

File: src/hunter/const.py [hunk 1]
```diff
@@ -2,14 +2,23 @@
...
+        try:
+            SITE_PACKAGES_PATHS.add(sysconfig.get_path(name, scheme))
+        except KeyError:
+            pass
 if hasattr(site, 'getsitepackages'):
     SITE_PACKAGES_PATHS.update(site.getsitepackages())
 if hasattr(site, 'getusersitepackages'):
     SITE_PACKAGES_PATHS.add(site.getusersitepackages())
-SITE_PACKAGES_PATHS.add(get_python_lib())
+if sys.version_info >= (3, 10):
```
Comment (The-Compiler) [inline src/hunter/const.py:18]: Are you sure you mean Python 3.10+ here?

File: src/hunter/const.py [hunk 2]
```diff
@@ -2,14 +2,23 @@
...
+        try:
+            SITE_PACKAGES_PATHS.add(sysconfig.get_path(name, scheme))
+        except KeyError:
+            pass
 if hasattr(site, 'getsitepackages'):
     SITE_PACKAGES_PATHS.update(site.getsitepackages())
 if hasattr(site, 'getusersitepackages'):
     SITE_PACKAGES_PATHS.add(site.getusersitepackages())
-SITE_PACKAGES_PATHS.add(get_python_lib())
+if sys.version_info <= (3, 10):
```
Comment (The-Compiler) [inline src/hunter/const.py:18]:
```suggestion
if sys.version_info < (3, 10):
```

since it's deprecated in 3.10 :wink:

**Comment by ionelmc** [comment]
Dooh! 🤦‍♂️

**Comment by The-Compiler** [comment]
I just found [site.getsitepackages()](https://docs.python.org/3/library/site.html#site.getsitepackages) and wonder if it does what you want?

```pycon
$ python3.10 -Wall
Python 3.10.0b1 (default, May  4 2021, 14:26:39) [GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from distutils.sysconfig import get_python_lib
<stdin>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<stdin>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
>>> get_python_lib()
'/usr/lib/python3.10/site-packages'
>>> from site import getsitepackages
>>> getsitepackages()
['/usr/lib/python3.10/site-packages']
```

seems to work on e.g. Ubuntu 20.04 too:

```pycon
>>> import site
>>> site.getsitepackages()
['/usr/local/lib/python3.8/dist-packages', '/usr/lib/python3/dist-packages', '/usr/lib/python3.8/dist-packages']
```

**Comment by ionelmc** [comment]
Yes, I know about it - it's actually used:
```py
if hasattr(site, 'getsitepackages'):
    SITE_PACKAGES_PATHS.update(site.getsitepackages())
```

But I have chosen to not solely rely on it because everyone messes with that site.py file (distro customizations, virtualenv implementing workarounds or whatever). It wouldn't include distro-packages for sure.

**Pull request merged**